src/app/detail/ui/ActivityWrapper.tsx

```diff
@@ -9,30 +9,102 @@ import { ActivityType } from '@prisma/client'
-import { useEffect, useState } from 'react'
+import useSWR, { useSWRConfig } from 'swr'
+import { useSelector } from 'react-redux'
+import { selectTaskBoard } from '@/redux/features/taskBoardSlice'
+import { selectAuthDetails } from '@/redux/features/authDetailsSlice'
+import { generateRandomString } from '@/utils/generateRandomString'
+import { z } from 'zod'
+import { ClientResponseSchema, InternalUsersSchema } from '@/types/common'
 
 export const ActivityWrapper = ({ token, task_id }: { token: string; task_id: string }) => {
   const { data: activities, isLoading } = useSWR(`/api/tasks/${task_id}/activity-logs/?token=${token}`, fetcher, {
-    refreshInterval: 1000,
+    refreshInterval: 10_000,
   })
+  const { tokenPayload } = useSelector(selectAuthDetails)
+  const { assignee } = useSelector(selectTaskBoard)
+  const { mutate } = useSWRConfig()
+  const currentUserId = tokenPayload?.clientId ?? tokenPayload?.internalUserId
+  const currentUserDetails = z
+    .union([InternalUsersSchema, ClientResponseSchema])
+    .parse(assignee.find((el) => el.id === currentUserId))
 
-  const [isFirstPageLoad, setIsFirstPageLoad] = useState(true)
+  const cacheKey = `/api/tasks/${task_id}/activity-logs/?token=${token}`
 
-  const searchParams = useSearchParams()
-  const commentId = searchParams.get('commentId')
+  // Handle comment creation
+  const handleCreateComment = async (postCommentPayload: CreateComment) => {
+    const tempLog: LogResponse = {
+      id: generateRandomString('temp-comment'),
+      type: ActivityType.COMMENT_ADDED,
+      details: {
+        content: postCommentPayload.content,
+      },
+      taskId: task_id,
+      userId: currentUserId as string,
+      userRole: 'internalUser',
+      workspaceId: '',
+      initiator: {
+        status: '',
+        id: currentUserDetails?.id as string,
+        givenName: currentUserDetails?.givenName || '',
+        familyName: currentUserDetails?.familyName || '',
+        email: currentUserDetails?.email || '',
+        companyId: '',
+        avatarImageUrl: currentUserDetails?.avatarImageUrl || null,
+        customFields: {},
+        fallbackColor: currentUserDetails?.fallbackColor || null,
+        createdAt: currentUserDetails?.createdAt || new Date().toISOString(),
+        isClientAccessLimited: false,
+        companyAccessList: null,
+      },
+      createdAt: new Date().toISOString(),
+    }
+
+    const optimisticData = activities ? [...activities.data, tempLog] : [tempLog]
+
+    try {
+      await mutate(
+        cacheKey,
+        async () => {
+          // Post the actual comment to the server
+          await postComment(token, postCommentPayload)
+          // Return the actual updated data (this will trigger revalidation)
+          return await fetcher(cacheKey)
+        },
+        {
+          optimisticData: { data: optimisticData },
+          rollbackOnError: true,
+          revalidate: true, // Make sure to revalidate after mutation
```

Comment:
this is fine for now, not blocking merging this PR, but I dont think you have to do revalidate. This is what [populateCache](https://swr.vercel.app/docs/mutation#update-cache-after-mutation) is for.